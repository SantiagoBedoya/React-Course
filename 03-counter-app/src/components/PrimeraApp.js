import PropTypes from 'prop-types';

const PrimeraApp = ({saludo, subtitulo}) => {
  return (
    <>
      <h1><i>{saludo}</i></h1>
      <h3>{subtitulo}</h3>
    </>
  )
}
PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired
}
PrimeraApp.defaultProps = {
  subtitulo: 'Hola Perras'
}

export default PrimeraApp;