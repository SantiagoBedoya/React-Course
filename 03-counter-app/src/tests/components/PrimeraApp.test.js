import {shallow} from 'enzyme';
import PrimeraApp from "../../components/PrimeraApp";

describe('Pruebas component PrimeraApp', () => {
  // test('debe mostrar el mensaje "Hola, Soy Goku"', () => {
  //   const saludo = 'Hola, soy goku';
    
  //   const {getByText} = render(<PrimeraApp saludo={saludo} />)

  //   expect(getByText(saludo)).toBeInTheDocument();
  // })
  

  test('debe de mostrar <PrimeraApp /> correctamente', () => {
    const wrapper = shallow(<PrimeraApp saludo="Perra" />)
    expect(wrapper).toMatchSnapshot();
  })

  test('debe de mostrar <PrimeraApp /> con subititulo', () => {
    const wrapper = shallow(
      <PrimeraApp 
        saludo="Perra"
        subtitulo="Maricones"
      />
    )
    const textParrafo = wrapper.find('h3').text();
    expect(textParrafo).toBe('Maricones');
  })
  

})
