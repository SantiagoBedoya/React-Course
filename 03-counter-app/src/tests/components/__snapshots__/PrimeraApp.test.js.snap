// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas component PrimeraApp debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    <i>
      Perra
    </i>
  </h1>
  <h3>
    Hola Perras
  </h3>
</Fragment>
`;
